### ISSUE TITLE
Dev Click Branch

### ISSUE THREAD

**Issue by stackpoint** [TheDivexz/excelmakesbelhappypack#11]
See: https://github.com/TheDivexz/excelmakesbelhappypack/tree/dev-click

Ran some test games. It looks like the dev clicks are working correctly. 

However, I was definitely suffering from performance problems in this branch compared to the master branch. @mellester have you noticed this?

**Comment by mellester** [comment]
i have not noticed. i might make the dev click bi-yearly then  and mayby run the profiler

**Comment by stackpoint** [comment]
Did you make any progress with this? I found a lot of target errors in the error.log files. About 600,000 lines of this:
```
[gamestate.cpp:7179]: Undefined event target tooltip: target
[eventtarget.cpp:341]: Undefined event target: target
```

I also did a bit of testing and I found that this commit doesn't have any performance problems: https://github.com/TheDivexz/excelmakesbelhappypack/tree/8d2aac6222c4f588def7e572f19d3f4455c4326f

This is the commit before you added the "Build a min into devclick" commit and the two others afterwards.

**Comment by mellester** [comment]
Hmm weird, the min into devclick is a simple small change. The 2 after that however was a mison expanded fix . As in I moved sthe startup events of ME so it would trigger on startup.
And moved the a the on_develment to a singel file

**Comment by stackpoint** [comment]
I tried to find the profiler but it is no longer there and found [this](https://forum.paradoxplaza.com/forum/index.php?threads/eu4_profiling-exe-is-gone-after-1-29-patch.1257121/) instead.

This is the [error.log](https://github.com/TheDivexz/excelmakesbelhappypack/files/4474533/error.log) file of the game running for a couple of days on the [dev click branch](https://github.com/TheDivexz/excelmakesbelhappypack/tree/dev-click). As you can see there is a cascade of errors as the game is running which is probably causing the problem.

The three commits I mentioned earlier is almost certainly the cause of the problem because the prior commit did not have those errors either.

**Comment by mellester** [comment]
Might indeed be the case. I was unable to pin down the problem but somehow manged to get a commit to stop giving erros

**Comment by stackpoint** [comment]
I don't think it has to be separated into specific adm/dip/mil ideas. You can wait 30 days for the AI to choose the groups and then add ideas one by one until you run out of total ideas.

**Comment by stackpoint** [comment]
Had a chance to test [this](https://www.dropbox.com/s/0sixndm5mjlbtjq/HotMess.7z?dl=0) from the discord.

I wasn't able to get the catch and release ideas working at all. Was there a certain method I'm missing?

Seems like there are still performance issues with the dev click (?); 15 years at speed 5 with your branch took me 15 minutes while the original branch only took 10 minutes.

Still getting thousands of the following error message in the error.log file:
```
[gamestate.cpp:7139]: Undefined event target: DeVCostTarget
[gamestate.cpp:7179]: Undefined event target tooltip: DeVCostTarget
[eventtarget.cpp:341]: Undefined event target: DeVCostTarget
```

**Comment by stackpoint** [comment]
I tested the new branch by releasing the country entirely instead of vassalizing them and it gave them the ideas for Economic.

However the dev click button does not seem to be working.

**Comment by stackpoint** [comment]
Performance still poor. 11 years in 10 minutes compared to the following:

> Seems like there are still performance issues with the dev click (?); 15 years at speed 5 with your branch took me 15 minutes while the original branch only took 10 minutes.

There also seems to be a problem with the institution tech penalty now. It's not calculating the correct amount now. Looks like the on_startup action is overwriting the institution check date.

**Comment by mellester** [comment]
In which case i Might jsut turn it into a random dev province

**Comment by stackpoint** [comment]
From https://github.com/TheDivexz/excelmakesbelhappypack/pull/26:

I've been doing all my testing of the dev click on an earlier commit on this branch because that was the commit I knew was working. I added a fix for the lag but there are some other problems.

Apparently, it seems like sometimes the AI deliberately hits the monarch point cap before taking tech early or at all. I don't know if this is a bug introduced in the dev click code (although I don't know how we could have affecting monarch point behavior). But it seems like the AI gets stuck not spending their points and hitting the cap.

Since the code just checks if the AI has hit the cap without checking if they can take tech means that the end result of this code being Mantua by the year 1492, spending 3520 monarch points (1787 military) developing their capital while being two military tech behind.

This doesn't happen to all countries, not sure what the pattern is and if the dev code is even working on the master branch.

A couple things:

- Delaying the point deduction/dev event after the AI clicks by 1 month so the AI has a chance to sit at the monarch cap (giving them the chance to take tech)
- Setting ai_importance = 1, to allow the AI to do other queued activities before clicking (don't think this works)
- Having the institution counter tick in December/February instead of the 1st of January since the AI tends to evaluate/take tech on the new year. (game is less burdened, AI has chance to evaluate tech)

I did some testing with the dev click on the current branch. Institution penalty is broken in the trigger but not in the tooltip.
